crates/libcontainer/src/container/builder.rs

```diff
@@ -117,11 +122,19 @@ impl<'a> ContainerBuilder<'a> {
     /// # use libcontainer::syscall::syscall::create_syscall;
     ///
     /// ContainerBuilder::new("74f1a4cb3801".to_owned(), create_syscall().as_ref())
-    /// .with_pid_file(Some("/var/run/docker.pid"));
+    /// .with_pid_file(Some("/var/run/docker.pid")).expect("invalid pid file");
     /// ```
-    pub fn with_pid_file<P: Into<PathBuf>>(mut self, path: Option<P>) -> Self {
-        self.pid_file = path.map(|p| p.into());
-        self
+    pub fn with_pid_file<P: Into<PathBuf>>(mut self, path: Option<P>) -> Result<Self> {
+        self.pid_file = if let Some(p) = path {
+            Some(
+                p.into()
+                    .canonicalize()
+                    .context("failed canonicalize pid file path")?,
```

Comment:
```suggestion
                    .with_context(|| format!("failed canonicalize pid file path"))?,
```

Reply:
Did you mean to add the `path` here too? Adjusted it similarly to the other one.